include/sg14/bits/fixed_point_math.h

```diff
@@ -0,0 +1,142 @@
+
+//          Copyright Timo Alho 2016.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file ../LICENSE_1_0.txt or copy at
+//          http://www.boost.org/LICENSE_1_0.txt)
+
+/// \file
+/// \brief some cmath specializations for `sg14::fixed_point` type;
+
+#ifndef FIXED_POINT_MATH_H_
+#define FIXED_POINT_MATH_H_
+
+#include <sg14/fixed_point.h>
+#include <type_traits>
+
+namespace sg14 {
+
+    namespace _fixed_point_impl {
+
+        template<class FixedPoint>
+        constexpr FixedPoint rounding_conversion(double d) {
+            using one_longer = make_fixed<FixedPoint::integer_digits, FixedPoint::fractional_digits + 1>;
```

Comment:
It's only used actually at compile time: rounding the coefficients of the polynomial correctly gives a pretty significant improvement (about 2 bits) for the result, so I use this to convert from the `double` literals. So this can be removed if one instead inputs the coefficients as integers and uses `from_data`, but that does make adjusting things more difficult. It could also work to export the coefficients from Mathematica as decimal values already rounded to the closest representable integer.

Note though that the evaluation of the polynomial does depend crucially on promoting to a wider type (as the multiplications would otherwise overflow), so without *a lot* more trickery, it's not possible to evaluate `exp` if there's no wider type available.

P.S.: Sorry about my silence on the arithmetic operator semantics, I haven't had any time to allocate for the project where I'm using these for a while. Unfortunately that situation won't really improve at least for the next couple of weeks, either.